### PR TITLE
[CSL-1980] Add pre_filter_expression to search/browse

### PR DESCRIPTION
**Pull request by stanlp1**

### CODE REVIEW

**Review by mocca102 — APPROVED** [review]
LGTM! let's have someone else have a look too

File: src/Constructorio_NET.Tests/client/modules/BrowseTests.cs [hunk 1]
```diff
@@ -75,6 +74,48 @@ public async Task GetBrowseResultsWithFilters()
...
+                    },
+                    {
+                    and:
+                        [
+                        { name: 'Color', value: 'blue' },
+                        { name: 'Brand', value: 'XYZ' },
+                    ],
+                    },
+                ],
+            }");
```
Comment (Mudaafi) [inline src/Constructorio_NET.Tests/client/modules/BrowseTests.cs:97]: hmm, I have some ideas on this but I'm not sure which is the better approach. On one hand, this is pretty clear cut as long as the consumer formats it right, on the other, it's pretty against the strict-type style of C# :/ Could we pair on this for a bit to exchange ideas?

**Comment by stanlp1** [comment]
@Mudaafi This is looking good. Let's add some not and range expressions to the tests, also lets add a test that has multiple expression types on a single level. Should work fine but would be nice to have it as a test.

**Comment by stanlp1** [comment]
This is looking good to me. I can't approve this since I opened the pr but let's have one more person take a look.

**Review by crgee1 — APPROVED** [review]
lgtm

**Pull request merged**